mutex.go

```diff
@@ -91,15 +73,31 @@ func (m *mutex) Lock() {
 		if err == nil {
 			res.Body.Close()
 			if res.StatusCode == 200 {
-				return
+				// Check for timeout before returning.
+				okToReturn = true
```

Comment:
With this approach on a success response from GCS, you seem to be waiting `time.Duration(i) * time.Millisecond` duration unconditionally, thus delaying the method return even if the lock has been acquired potentially "long" time ago.

Why is checking timeout needed on success? Also, not sure what "check for timeout" means :) If it's about the context, then they say "context is done" because the context could also be cancelled, so multiple reasons, not just a context timeout expiration.